### ISSUE TITLE
Visual Studio Code does not change when High Contrast is turned on

### ISSUE THREAD

**Issue by alexandrudima** [Microsoft/vscode#43750]
(Moved from internal Windows Feedback bug database)

1. Turn on High Contrast Black
2. Launch VS Code
3. Notice that inside of the frame, the text shows up normally (not in high contrast). This is an accessibility issue

**Comment by alexandrudima** [comment]
The current workaround is to manually switch VS Code to the High Contrast Theme via `File > Preferences > Color Theme > High Contrast`

**Comment by egamma** [comment]
@bpasero is there electron API to detect high contrast mode automatically?

**Comment by bpasero** [comment]
Yeah I thought we detect high contrast on Windows, I have to check.

**Comment by bpasero** [comment]
@aeschli I see that the `hc-black` theme is correctly passed on to the renderer ([here](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/electron-browser/bootstrap/preload.js#L23)) but then is reset back to the previous one (e.g. `vs-black`). Can you check why the theme is not applied?

To reproduce:
* enable HC theme on Windows
* start VS Code
* expect it to show in HC theme

**Comment by aeschli** [comment]
@bpasero The hc theme needs to be set in the settings or on the theme service, like ([here](https://github.com/Microsoft/vscode/blob/03db54e4dd2fdca2b6ac4afaa3ac6a97a637c7c1/src/vs/workbench/electron-browser/window.ts#L220). This will trigger all theming participants that then ally the themeing and generate the css rules.
Setting the class name just helps with the initial window appearance.

**Comment by bpasero** [comment]
@aeschli it looks like you removed that code that handles high contrast on Windows and never added it back to the renderer (https://github.com/Microsoft/vscode/commit/f938ba6e1b758727bcead0ef9f42a47949007ac9#diff-da74529a2453ded5da85bf71e4a75a1c)?

I created https://github.com/Microsoft/vscode/pull/44176 to prefer the HC theme if we detect high contrast mode. It is a bit ugly though because the settings are still containing the previous theme, not sure we should just blindly overwrite it because the HC theme on Windows could be a temporary thing.
